types/ibc.go

```diff
@@ -0,0 +1,107 @@
+package types
+
+type IBCEndpoint struct {
+	PortId    string `json:"port_id"`
+	ChannelId string `json:"channel_id"`
+}
+
+type IBCChannel struct {
+	Endpoint             IBCEndpoint `json:"endpoint"`
+	CounterpartyEndpoint IBCEndpoint `json:"counterparty_endpoint"`
+	Order                IBCOrder    `json:"order"`
+	Version              string      `json:"version"`
+	// optional
+	CounterpartyVersion string `json:"counterparty_version,omitempty"`
+	ConnectionID        string `json:"connection_id"`
+}
+
+// TODO: test what the sdk Order.String() represents and how to parse back
+// Proto files: https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/ibc/core/channel/v1/channel.proto#L69-L80
+// Auto-gen code: https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/x/ibc/core/04-channel/types/channel.pb.go#L70-L101
+type IBCOrder = string
+
+// These are the only two valid values for IbcOrder
+const Unordered = "ORDER_UNORDERED"
+const Ordered = "ORDER_ORDERED"
+
+// IBCTimeoutHeight Height is a monotonically increasing data type
+// that can be compared against another Height for the purposes of updating and
+// freezing clients.
+// Ordering is (revision_number, timeout_height)
+type IBCTimeoutHeight struct {
+	// the version that the client is currently on
+	// (eg. after reseting the chain this could increment 1 as height drops to 0)
+	RevisionNumber uint64 `json:"revision_number"`
+	// block height after which the packet times out.
+	// the height within the given revision
+	TimeoutHeight uint64 `json:"timeout_height"`
+}
+
+type IBCPacket struct {
+	Data             []byte           `json:"data"`
+	Src              IBCEndpoint      `json:"src"`
+	Dest             IBCEndpoint      `json:"dest"`
+	Sequence         uint64           `json:"sequence"`
+	TimeoutHeight    IBCTimeoutHeight `json:"timeout_height"`
```

Comment:
Req: Should be a pointer as it is either height or timestamp for the timeout

Reply:
:100: see above